models/migrate.go

```diff
@@ -83,6 +83,13 @@ func insertIssue(ctx context.Context, issue *Issue) error {
 		}
 	}
 
+	if issue.ForeignReference != nil {
+		issue.ForeignReference.LocalID = issue.Index
```

Comment:
It's inconsistent with above code. See https://github.com/go-gitea/gitea/pull/18446/files#r804317877

Reply:
Good catch, it is inconsistent. The tests did not fail because they have a false positive (ID == Number). A new commit was added to fix this inconsistency and make it so there cannot be a false positive in the test. I also re-read the entire commit and replace one last hardcoded "issue" with ForeignTypeIssue.

---
[source](https://lab.fedeproxy.eu/forgefriends/forgefriends/-/merge_requests/36#note_1043)